collectors/alibaba/collector.js

```diff
@@ -0,0 +1,314 @@
+            paginate: 'Pages'
+        }
+    },
+    RDS: {
+        DescribeDBInstances: {
+            property: 'Items',
+            subProperty: 'DBInstance',
+            apiVersion: '2014-08-15',
+            paginate: 'Pages'
+        }
+    },
+    POLARDB: {
+        DescribeDBClusters: {
+            property: 'Items',
+            subProperty: 'DBCluster',
+            apiVersion: '2017-08-01',
+            paginate: 'Pages'
+        }
+    }
+};
+
+var postcalls = [
+    {
+        ECS: {
+            DescribeInstanceStatus: {
+                reliesOnService: 'ecs',
+                reliesOnCall: 'DescribeInstances',
+                filterKey: ['InstanceId'],
+                filterValue: ['InstanceId'],
+                resultKey: 'InstanceId',
+                apiVersion: '2014-05-26'
+            }
+        },
+        RAM: {
+            GetPolicy: {
+                reliesOnService: 'ram',
+                reliesOnCall: 'ListPolicies',
+                filterKey: ['PolicyName', 'PolicyType'],
+                filterValue: ['PolicyName', 'PolicyType'],
+                resultKey: 'PolicyName',
+                resultFilter: 'DefaultPolicyVersion'
+            },
+            GetUser: {
+                reliesOnService: 'ram',
+                reliesOnCall: 'ListUsers',
+                filterKey: ['UserName'],
+                filterValue: ['UserName'],
+                resultKey: 'UserName',
+                resultFilter: 'User'
+            }
+        }
+    }
+];
+
+var collect = function(AlibabaConfig, settings, callback) {
+    if (settings.gather) {
+        return callback(null, calls, postcalls);
+    }
+
+    var regions = helpers.regions(settings);
+
+    var collection = {};
+
+    async.eachOfLimit(calls, 10, function(call, service, serviceCb) {
+        var serviceLower = service.toLowerCase();
+        if (!collection[serviceLower]) collection[serviceLower] = {};
+
+        async.eachOfLimit(call, 15, function(callObj, callKey, callCb) {
+            if (settings.api_calls && settings.api_calls.indexOf(service + ':' + callKey) === -1) return callCb();
+            if (!collection[serviceLower][callKey]) collection[serviceLower][callKey] = {};
+
+            var callRegions = regions[serviceLower];
+            var requestOption = { method: callObj.method || 'POST' };
+
+            async.eachLimit(callRegions, helpers.MAX_REGIONS_AT_A_TIME, function(region, regionCb) {
+                if (settings.skip_regions &&
+                    settings.skip_regions.indexOf(region) > -1 &&
+                    globalServices.indexOf(service) === -1) return regionCb();
+                if (!collection[serviceLower][callKey][region]) collection[serviceLower][callKey][region] = {};
+
+                var LocalAlibabaConfig = JSON.parse(JSON.stringify(AlibabaConfig));
+
+                var endpoint = `https://${serviceLower}.aliyuncs.com`;
+                LocalAlibabaConfig['endpoint'] = endpoint;
```

Comment:
do we use the variable endpoint anywhere else? if not can we just assign it directly